UICAL-148 use react-intl for date formatting

zburke opened the pull request:
## Purpose

Some date display was pretty convoluted, storing formatters as translations and passing those to moment and passing that to `dateformat` and ... yeesh. 

## Approach

Use `react-intl` externally to present values to the UI, i.e. don't call `moment` functions for display. This resolves date strings like `01/20/AMAMAMAM` which passed the `dateFormat` translation value as an argument to `moment.format()`. This also allows the `dateformat` dependency to be removed completely. 

Additionally, use the localized value for `Closed`. It was always available, but we just weren't using it.

Finally, add `react-hot-loader` as a dev-dep and update `index.js` to support hot-loading. This is a standard practice for other modules that was missing here, but arguably that should be handled in a separate PR; LMK if you would prefer that. I didn't actually intend to include it here but forgot to split the changes on `package.json`. 

Refs [UICAL-148](https://issues.folio.org/browse/UICAL-148)

zburke committed:
UICAL-148 use react-intl for date formatting

Some date display was pretty convoluted, storing formatters as
translations and passing those to moment and passing that to
`dateformat` and ... yeesh. It's like this: use `moment` internally to
store date values; use `react-intl` externally to present values to the
UI, i.e. don't call `moment.format()` for display. This resolves date
strings like `01/20/AMAMAMAM` which passed the `dateFormat` translation
value as an argument to `moment.format()`.

Additionally, use the localized value for `Closed`. It was always
available, but we just weren't using it.

Refs UICAL-148

Dmitriy-Litvinenko commented on translations/ui-calendar/en.json:
```diff
@@ -69,7 +69,6 @@
   "regularLibraryHoursCalendar":"Regular Library Hours:",
   "selectTemplate":"Select Template",
   "copy":"Copy Previous",
-  "dateFormat":"MM/DD/YYYY",
```
Used in ui-calendar\stripes\ui-calendar\test\bigtest\helpers\messageConverters.js
We should change formatDateString before deleting dateFormat
Just for example `return date.format('MM/DD/YYYY');`

Dmitriy-Litvinenko commented on src/settings/ServicePointDetails.js:
```diff
@@ -207,7 +197,13 @@ class ServicePointDetails extends React.Component {
                   data-test-next-period-item-label
                 >
-                  {this.state.currentPeriod.startDate + ' - ' + this.state.currentPeriod.endDate + ' (' + this.state.currentPeriod.name + ')'}
+                  {this.state.currentPeriod.startDate}
```
We have test error here "AssertionError: expected '7/25/2021-8/29/2021 (test)' to equal '07/25/2021 - 08/29/2021 (test)'"

zburke committed:
restore whitespace that stupid lint rules caused me to remove

zburke committed:
lint

zburke committed:
separate date from display-date

There is not great separation between display-date format, e.g. M/D/YYYY
for the locale `en-US` and the internal formatting of dates, e.g.
`MM/DD/YYYY` in the locale `en-US`. The adjustments here attempt to sync
that up, looking for the former when dates are displayed as text but
using the latter when they are input as values in forms.

zburke committed:
debugging output cleanup
